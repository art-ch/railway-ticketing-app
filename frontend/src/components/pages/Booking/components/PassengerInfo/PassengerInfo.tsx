import React from 'react';

import { PassengerDetails } from '@/models/passengerDetails.model';

export type PassengerInfoProps = { passengerInfo: PassengerDetails };

export const PassengerInfo = ({ passengerInfo }: PassengerInfoProps) => {
  const { name, email, phone } = passengerInfo;

  return (
    <div>
      <p className="font-bold pb-1">Passenger Info</p>

      <table className="w-full">
        <thead>
          <tr className="text-sm font-semibold">
            <th className="text-left min-w-45">Name</th>
            <th className="text-left min-w-45">Email</th>
            <th className="text-left min-w-45">Phone</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td className="min-w-45">{name}</td>
            <td className="min-w-45">{email}</td>
            <td className="min-w-45">{phone}</td>
          </tr>
        </tbody>
      </table>
    </div>
  );
};
