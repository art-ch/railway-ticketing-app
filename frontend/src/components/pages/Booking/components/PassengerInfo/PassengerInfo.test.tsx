import { render } from '@testing-library/react';
import { PassengerInfo } from './PassengerInfo';
import { PassengerDetails } from 'railway-ticketing-app-sdk';

describe('Passenger Info component', () => {
  it('should render correctly', () => {
    const passengerInfo = {
      name: 'gdjskalg',
      email: 'fgdjsalkg@gmail.com',
      phone: '5784397654645'
    } as PassengerDetails;

    const { container } = render(
      <PassengerInfo passengerInfo={passengerInfo} />
    );

    expect(container).toMatchSnapshot();
  });
});
