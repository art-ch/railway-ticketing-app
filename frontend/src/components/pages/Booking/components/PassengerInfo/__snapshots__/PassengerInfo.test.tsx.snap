// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passenger Info component should render correctly 1`] = `
<div>
  <div>
    <p
      class="font-bold pb-1"
    >
      Passenger Info
    </p>
    <table
      class="w-full"
    >
      <thead>
        <tr
          class="text-sm font-semibold"
        >
          <th
            class="text-left min-w-45"
          >
            Name
          </th>
          <th
            class="text-left min-w-45"
          >
            Email
          </th>
          <th
            class="text-left min-w-45"
          >
            Phone
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="min-w-45"
          >
            gdjskalg
          </td>
          <td
            class="min-w-45"
          >
            fgdjsalkg@gmail.com
          </td>
          <td
            class="min-w-45"
          >
            5784397654645
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
